source/language/delays.rst

```diff
@@ -127,23 +128,43 @@ these side effects. Also contrary to TeX, we prohibit overlapping gates.
 Operations on durations
 -----------------------
 
-We can add/subtract two durations, or multiply or divide them by a constant, to get a new
-duration. Division of two durations results in a machine-precision float 
-(see :ref:`divideDuration`). Negative durations are allowed, however
-passing a negative duration to a ``gate[duration]`` or ``box[duration]`` expression will result in an error.
-All operations on durations happen at compile time since ultimately all
-durations, including stretches, will be resolved to constants.
+We can add/subtract two durations, or multiply or divide them by a
+constant, to get a new duration. Division of two durations results in
+a machine-precision float (see :ref:`divideDuration`). Negative
+durations are allowed, however passing a negative duration to a
+``gate[duration]`` or ``box[duration]`` expression will result in an
+error.  All operations on ``const`` durations, including stretches,
+can happen at compile time. Note that some programs may make use of
+non-const duration (e.g. as ``input`` parameters or in
+backend-specific externs).
 
 .. code-block::
 
-       duration a = 300ns;
-       duration b = durationof({x $0;});
+       const duration a = 300ns;
+       const duration b = durationof({x $0;});
        stretch c;
        // stretchy duration with min=300ns
        stretch d = a + 2 * c;
        // stretchy duration with backtracking by up to half b
        stretch e = -0.5 * b + c;
 
+Because `stretch` introduces a `const duration`, initialization with a non-`const` expression are a type error. On the other hand, because stretch resolution requires solving a system of constraints, a compiler may reject a well-typed program involving stretch durations due to a failure to resolve the constraint system.
+
+.. code-block::
+
+       input duration a; // not const
+       const duration b = 300ns;
+       stretch c;
+
+       // type error: const initializer expected
+       stretch d = c + a;
+
+       // stretch resolution error: 'e' resolved to -200ns
+       stretch e = b - 500ns;
```

Comment:
Imo this is not an error - it's just an ineffective bound because it's strictly weaker than the implicit `e > 0` bound that exists for all stretches.